clang/lib/Analysis/CFG.cpp

```diff
@@ -1104,6 +1102,27 @@ class CFGBuilder {
     }
   }
 
+  TryResult analyzeLogicOperatorCondition(BinaryOperatorKind Relation,
```

Comment:
Afaik, `APFloat` has operator overloads for `==` and friends; can we move the implementation of this (and of the overload of `analyzeLogicOperatorCondition()` above) into a template, i.e.
```c++
  template <typename APFloatOrInt>
  TryResult analyzeLogicOperatorCondition(BinaryOperatorKind Relation,
                                          const APFloatOrInt &Value1,
                                          const APFloatOrInt &Value2)
```
and reuse it for both floats and ints (yes, we’d be dropping the `Value1.isSigned() == Value2.isSigned()` assertion in the `APSInt` case, but I think that’s not too big of a deal in this case because this is only called in a context where we check for that beforehand anyway).